### PR TITLE
vterrors cleanup

### PR DESCRIPTION
**Pull request by sougou**
## Summary of changes
* Introduced new vtrpcpb.Code to match grpc codes.
* Cleaned and simplified vterrors. Renamed all functions to have idiomatic names.
* vterrors Aggregate function priorities have been re-adjusted to better reflect reality.
* vtrpcpb.RPCError: The old error field has been renamed to legacy_code. The new field is called Code.
* Functions that convert to and from RPCError fallback to legacy code if new code is absent. Additionally, legacy `INTERNAL` error maps to `UNAVAILABLE` in the new world.
* Remapped error codes to the more appropriate ones. Specifically, FailedPrecondition is a reliable indicator that a failover may be in progress. Many MySQL codes have been mapped to canonical codes. This will remain an ongoing process.
* IsConnErr has been moved to mysqlconn/constants.go.
* All error types have been consolidated to use vterrors.vtError. The only exception is sqldb.SQLError. Along with this, TabletError prefixes like `not_in_tx` are gone.
* `/debug/vars` for vttablet and vtgate exports only one `Errors` variable, with subcategories like `INVALID_ARGUMENT`, etc. In the case of vtgate it's a fourth dimension. This may have to be revisited.
* `queryservice.Wrap` has been improved. It now tells the wrapper if a request can be retried. This is required for StreamExecute because it should not allow retry if results have already been sent.
* VTGate's logging decision is driven by the error code rather than client vs server-side error.

### CODE REVIEW

**Comment by sougou** [comment]
This part is now ready for review. This change is the backward compatible part of the error handling changes. The subsequent changes will involve recategorization of errors and how they're handled.

**Comment by sougou** [comment]
@alainjobart @michael-berlin This change is now ready for review.

File: go/vt/vterrors/vterrors.go [hunk 1]
```diff
@@ -2,138 +2,45 @@ package vterrors
...
-	}
-	if vtError, ok := in.(VtError); ok {
-		return &VitessError{
-			Code:    vtError.VtErrorCode(),
-			err:     in,
-			Message: fmt.Sprintf("%s%s", prefix, in.Error()),
-		}
-	}
-	return fmt.Errorf("%s%s", prefix, in)
+func (e *vtError) Error() string {
```
Comment (alainjobart) [inline go/vt/vterrors/vterrors.go]: The comment is not accurate any more here.

File: proto/vtrpc.proto [hunk 2]
```diff
@@ -33,95 +33,213 @@ message CallerID {
...
 
-// ErrorCode is the enum values for Errors. Internally, errors should
-// be created with one of these codes. These will then be translated over the wire
-// by various RPC frameworks.
-enum ErrorCode {
-  // SUCCESS is returned from a successful call.
-  SUCCESS = 0;
-
-  // CANCELLED means that the context was cancelled (and noticed in the app layer,
+// Code represnts canonical error codes. The names, numbers and comments
```
Comment (alainjobart) [inline proto/vtrpc.proto]: represnts -> represents

File: proto/vtrpc.proto [hunk 3]
```diff
@@ -33,95 +33,213 @@ message CallerID {
...
+  // DATA_LOSS indicates unrecoverable data loss or corruption.
+  DATA_LOSS = 15;
+}
+
+// LegacyErrorCode is the enum values for Errors. This type is deprecated.
+// Use Code instead. Background: In the initial design, we thought
+// that we may end up with a different list of canonical error codes
+// than the ones defined by grpc. In hindisght, we realize that
+// the grpc error codes are fairly generic and mostly sufficient.
+// In order to avoid confusion, thie type will be deprecated in
```
Comment (alainjobart) [inline proto/vtrpc.proto]: thie -> this

**Pull request merged**